tests/snuba/api/endpoints/test_project_event_details.py

```diff
@@ -0,0 +1,182 @@
+        assert response.data['nextEventID'] == six.text_type(self.next_event.event_id)
+        assert response.data['previousEventID'] == six.text_type(self.prev_event.event_id)
+        assert response.data['groupID'] == six.text_type(self.cur_event.group.id)
+
+        # Same event can be looked up by primary key
+        url = reverse(
+            'sentry-api-0-project-event-details',
+            kwargs={
+                'event_id': self.cur_event.id,
+                'project_slug': self.cur_event.project.slug,
+                'organization_slug': self.cur_event.project.organization.slug,
+            }
+        )
+        response = self.client.get(url, format='json', data={
+            'enable_snuba': '1',
+        })
+
+        assert response.status_code == 200, response.content
+        assert response.data['id'] == six.text_type(self.cur_event.event_id)
+        assert response.data['nextEventID'] == six.text_type(self.next_event.event_id)
+        assert response.data['previousEventID'] == six.text_type(self.prev_event.event_id)
+        assert response.data['groupID'] == six.text_type(self.cur_event.group.id)
+
+    def test_snuba_no_prev(self):
+        options.set('snuba.events-queries.enabled', True)
+        url = reverse(
+            'sentry-api-0-project-event-details',
+            kwargs={
+                'event_id': self.prev_event.event_id,
+                'project_slug': self.prev_event.project.slug,
+                'organization_slug': self.prev_event.project.organization.slug,
+            }
+        )
+        response = self.client.get(url, format='json', data={
+            'enable_snuba': '1'
+        })
+
+        assert response.status_code == 200, response.content
+        assert response.data['id'] == six.text_type(self.prev_event.event_id)
+        assert response.data['previousEventID'] is None
+        assert response.data['nextEventID'] == self.cur_event.event_id
+        assert response.data['groupID'] == six.text_type(self.prev_event.group.id)
+
+    def test_snuba_with_environment(self):
+        options.set('snuba.events-queries.enabled', True)
+        url = reverse(
+            'sentry-api-0-project-event-details',
+            kwargs={
+                'event_id': self.cur_event.event_id,
+                'project_slug': self.cur_event.project.slug,
+                'organization_slug': self.cur_event.project.organization.slug,
+            }
+        )
+        response = self.client.get(url, format='json', data={
+            'enable_snuba': '1',
+            'environment': ['production', 'staging']
+        })
+
+        assert response.status_code == 200, response.content
+        assert response.data['id'] == six.text_type(self.cur_event.event_id)
+        assert response.data['previousEventID'] is None
+        assert response.data['nextEventID'] == self.next_event.event_id
+        assert response.data['groupID'] == six.text_type(self.prev_event.group.id)
```

Comment:
I think the test `test_event_ordering` that's marked `xfail` in the original tests can probably be copied here and it should work on the snuba version.

TLDR basically if we have events `a, b, c, d, e, f, g` with the same timestamp, and we are on event `b` then our postgres approach for `prev_event` ("select 5 events with a timestamp <= the current one") would fail because it would find `c, d, e, f, g` and miss `a`, but the snuba version shouldn't have that problem